types/dc/index.d.ts

```diff
@@ -63,34 +64,42 @@ declare namespace dc {
     }
 
     export interface Events {
-        trigger(fn: () => void, delay?: number): void;
+        trigger(closure: () => void, delay?: number): void;
     }
 
     export interface Errors {
         Exception(msg: string): void;
-        InvalidStateException(msg: string): void;
+        InvalidStateException(): void;
+        BadArgumentException(): void;
     }
 
     export interface Filter {
         isFiltered(value: any): boolean;
     }
 
     export interface Filters {
-        RangedFilter(low: any, high: any): Filter;
-        TwoDimensionalFilter(arr: Array<any>): Filter;
-        RangedTwoDimensionalFilter(arr: Array<any>): Filter;
+        RangedFilter(low: number, high: number): Array<number>;
```

Comment:
this seems incorrect, the return of these functions is an Array that has a isFiltered method, see the implementation:

the jsdoc annnotation in the code is incomplete...

/**
 * RangedFilter is a filter which accepts keys between `low` and `high`.  It is used to implement X
 * axis brushing for the {@link dc.coordinateGridMixin coordinate grid charts}.
 *
 * Its `filterType` is 'RangedFilter'
 * @name RangedFilter
 * @memberof dc.filters
 * @param {Number} low
 * @param {Number} high
 * @returns {Array<Number>}
 * @constructor
 */
dc.filters.RangedFilter = function (low, high) {
    var range = new Array(low, high);
    range.isFiltered = function (value) {
        return value >= this[0] && value < this[1];
    };
    range.filterType = 'RangedFilter';

    return range;
};

Reply:
@hansrwindhoff True, I just went off modifying the type definitions just based on jsdoc annotation and completely ignored the actual code. PTAL after my update.